src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,205 @@
+
+import com.puppycrawl.tools.checkstyle.internal.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.ModuleReflectionUtil;
+
+public class XdocsUrlTest {
+
+    private static final String PACKAGE_NAME =
+            "src/main/java/com/puppycrawl/tools/checkstyle/checks";
+
+    private static final String TREE_WORKER = "TreeWalker";
+
+    private static final String SUFFIX_CHECK = "Check";
+
+    private static final String CHECKS = "checks";
+
+    private static final String MISC = "misc";
+
+    private static final String ANNOTATION = "annotation";
+
+    private static final String COMMENTS_INDENTATION = "CommentsIndentation";
+
+    private static final String INDENTATION = "Indentation";
+
+    private static final String SUPPRESS_WARNINGS_HOLDER = "SuppressWarningsHolder";
+
+    private static final Path AVAILABLE_CHECKS_PATH = Paths.get("src/xdocs/checks.xml");
+
+    private static Map<String, ArrayList<String>> getXdocsMap() throws IOException {
+        final Map<String, ArrayList<String>> checksNamesMap = new HashMap<>();
+        final Set<Class<?>> checkSet = CheckUtil.getCheckstyleModules();
+        final Set<Class<?>> treeWalkerCheckSet = checkSet.stream()
+                .filter(ModuleReflectionUtil::isCheckstyleTreeWalkerCheck)
+                .collect(Collectors.toSet());
+        final Set<Class<?>> fileSetCheckSet = checkSet.stream()
+                .filter(ModuleReflectionUtil::isFileSetModule)
+                .collect(Collectors.toSet());
+        checkSet.clear();
+        checkSet.addAll(treeWalkerCheckSet);
+        checkSet.addAll(fileSetCheckSet);
+        for (Class<?> check : checkSet) {
+            final String checkName = check.getSimpleName();
+            if (!checkName.equals(TREE_WORKER)) {
+                String packageName = check.getPackage().getName();
+                packageName = packageName.substring(packageName.lastIndexOf('.') + 1);
+                if (packageName.equals(CHECKS)) {
+                    packageName = MISC;
+                }
+                if (checksNamesMap.get(packageName) == null) {
+                    final ArrayList<String> arrayList = new ArrayList<>();
+                    arrayList.add(checkName);
+                    checksNamesMap.put(packageName, arrayList);
+                }
+                else {
+                    checksNamesMap.get(packageName).add(checkName);
+                }
+            }
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest();
+        try (InputStream input = Files.newInputStream(AVAILABLE_CHECKS_PATH)) {
+            parser.parse(input, checkHandler);
+        }
+        final Map<String, ArrayList<String>> checksNamesMap = getXdocsMap();
```

Comment:
`ArrayList` => `List`
You should use the interface over the exact declaration when possible.

Reply:
done